CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+
+The other tricky case is where a transaction is phase-2 invalid. 
+To help us with this, we add two additional (phase-1) validity conditions inside the zone:
+
+1. (Safe collateral) All collateral inputs come from outputs outside the zone.
+2. (Excess collateral) The collateral required for a transaction containing scripts is increased: the base amount is now based on the fee for the transaction and _every transaction it depends on_ within the zone.
+    - “A depends on B” here means that A is required in order to determine B’s validity. Note that this means that for unresolved-input-resolving-output pairs the dependency is the opposite way around to the relationship for a normal input-output pair.
+
+If a transaction is phase-2 invalid, then the following occurs:
+
+- The node constructs an invalidity witness for the transaction. This consists of the invalid transaction, along with every transaction that it depends on from within the zone. 
+- The node posts the invalidity witness to the chain and claims the collateral associated with the failing transaction.
+
+## Rationale: how does this CIP achieve its goals?
+ 
+### Design
+
+#### Linearizability and value flow
+
+The requirement that validity zones must be linearizable ensures that we do not allow any value flow in the ledger that was not possible before. 
+Value always flows from inputs to outputs, and there are no loops. 
+We can see this because linearizability shows that we could produce a “normal” series of transactions by reordering the zone and translating unresolved inputs and resolving outputs into normal inputs and outputs.
+
+What we gain is the ability to specify and authorise this normal value flow in an out-of-order fashion. 
+Alice (whose transaction comes later in the value flow) can create their transaction first and send it to Bob (whose transaction comes earlier) and he can put them together into something valid.
+
+The linearizability requirement is the major driving difference between the design in this CIP and the one published in the "Babel Fees via Limited Liabilities" paper. 
+Both designs make use of the idea of “negative” inputs of one kind or another, but the liabilities’ design enables and embraces the possibility of loops in the value flow. 
+It is difficult to make this secure, which led to the painful restriction that minting policies be run in many cases where liabilities are used. 
+In practice that meant that many swap use cases could not be used with old tokens whose minting policies would not approve the new behaviour.
+
+Similarly, the liabilities design allowed easy access to arbitrary “flash loans” in Ada, which has often been a source of attacks on other blockchains.
+
+In contrast, if the value flow can be linearized then we know that nothing fundamentally new is possible here and we do not need to worry.
+
+#### Counterparty irrelevance
```

Comment:
I think we should also emphasize in this section the fact that two-way communication is not required